File: ruby-pattern/Adaptor/adaptor.rb

class Adaptee
  def initialize
  end

  def hoge
    "hoge"
  end

end

class Adaptor
  def initialize
    @adaptee = Adaptee.new()
  end

  def hoge
    "<H1>" + @adaptee.hoge + "</H1>"
  end


end

File: swift-pattern/Adaptor.playground/Sources/Adaptor.swift

public protocol NewAdaptor {
    func newMethod() -> String
}

public class Adaptor:Adaptee, NewAdaptor {

    public override init(){}

    public func newMethod()-> String{
        return "<H1>" + super.oldMethod() + "</H1>"
    }

}


public class Adaptee {

    public func oldMethod()->String{
        return "hoge"
    }
}
